nss-day-cohort-39/tabloid-mvc-depech-mode
# User Profiles List/Details

### Changes
-created User controller with Index, Details, and Re/Deactivate methods
-created user profile view model for user Index
-created corresponding repository methods
### Steps to test
1. Run the following **SQL** commands to set up test data

_add column to UserProfile table_
`ALTER TABLE UserProfile
ADD Active BIT DEFAULT 1 NOT NULL;`
_add dummy data to work with_
`SET IDENTITY_INSERT [UserProfile] ON
INSERT INTO [UserProfile] (
	[Id], [FirstName], [LastName], [DisplayName], [Email], [CreateDateTime], [ImageLocation], [UserTypeId])
VALUES (2, 'Steve', 'Brownlee', 'secondAdmin', 'steve@email.com', '06-14-2020', NULL, 1);
INSERT INTO [UserProfile] (
	[Id], [FirstName], [LastName], [DisplayName], [Email], [CreateDateTime], [ImageLocation], [UserTypeId], [Active])
VALUES (3, 'Loser', 'McDumb', 'lameeee', 'something@email.com', '06-14-2020', NULL, 1, 0);
SET IDENTITY_INSERT [UserProfile] OFF`

2. Go to the **Users** tab. The button near the top should read "show deactivated" and **the active users should be shown.**
3. Click the **Show Deactivated** button. The deactivated users should now be shown.
4. Check that the users **details** view works and that you can navigate back to Index.
5. Try **deactivating a use**r. Check to see that they moved to the corresponding view.
6. Try **reactivating a use**r.
7. Try **logging in with a deactivated user**.
